src/goto-programs/remove_virtual_functions.h

```diff
@@ -25,4 +25,37 @@ void remove_virtual_functions(
   const symbol_tablet &symbol_table,
   goto_functionst &goto_functions);
 
+/// Specifies remove_virtual_function's behaviour when the actual supplied
+/// parameter does not match any of the possible callee types
+enum class virtual_dispatch_fallback_actiont
+{
+  /// When no callee type matches, call the last passed function, which
+  /// is expected to be some safe default:
+  CALL_LAST_FUNCTION,
+  /// When no callee type matches, ASSUME false, thus preventing any complete
+  /// trace from using this path.
+  ASSUME_FALSE
+};
+
+class dispatch_table_entryt
+{
+ public:
+  dispatch_table_entryt() {}
```

Comment:
Use `= default`, even though I'm leaning towards just dropping that constructor and instead enforcing the use of a class id.